File: app/src/main/java/com/doapps/habits/models/IDayManager.java
package com.doapps.habits.models;

/**
 * Manages days changes
 */
public interface IDayManager {
    /**
     * Updates lists content by day of week
     * @param dayOfWeek Number of day from 1 to 7
     */
    void updateListByDayOfWeek(int dayOfWeek);

    /**
     * Updates lists content for today
     */
    void updateForToday();
}

File: app/src/main/java/com/doapps/habits/models/IDayManager.kt
package com.doapps.habits.models

/**
 * An interface for days change managers
 */
interface IDayManager {
  /**
   * Updates lists content by day of week
   * @param dayOfWeek Number of day from 1 to 7
   */
  fun updateListByDayOfWeek(dayOfWeek: Int)

  /**
   * Updates lists content for today
   */
  fun updateForToday()
}
